File: js/163248.js
/**
 * Lời giải từ bạn MaiTheDung
 */
function minCostRepairingCar(a, b) {
    const arrLength = a.length
    let count = cost = 0
    let indexOfSmallest
    for (let i = 0; i < arrLength; i++)
      if (a[i] < b[i]) {
        cost += a[i]
        count += 1
      }
      else {
        cost += b[i]
        count += 2
      }
    if (count === arrLength) {
      indexOfSmallest = b.indexOf(Math.min(...b))
      cost += b[indexOfSmallest] - a[indexOfSmallest]
    }
    if (count === arrLength * 2) {
      indexOfSmallest = a.indexOf(Math.min(...a))
      cost += a[indexOfSmallest] - b[indexOfSmallest]
    }
    return cost
  }

File: cs/163248.cs
// Lời giải từ bạn ChickenTapCode

long MinCostRepairingCar(int[] a, int[] b)
{
    long ret = 0;
    bool checka = false, checkb = false;
    int mincost = 1000000000;
    for(int i=0; i <a.Length; i++) {
        int min = Math.Min(a[i], b[i]);
        mincost = Math.Min(mincost, Math.Abs(a[i]- b[i]));
        if (min == a[i]) checka = true;
        if (min == b[i]) checkb = true;
        ret+= Math.Min(a[i], b[i]);
    }
    if (a.Length >1 && (!checka || !checkb)) ret+= mincost;
    return ret;
}
